### PR TITLE
SWC-135

### PR DESCRIPTION
**Pull request by KadenZipfel**

### CODE REVIEW

**Comment by blitz-1306** [comment]
This a bit reminding https://swcregistry.io/docs/SWC-104. Could there be other samples that are causing security issues by expression statements?

**Comment by KadenZipfel** [comment]
@blitz-1306 Yeah I can see how it's confusing using that example, especially since checking the return value should prevent this problem. I'll see if I can come up with some other examples.

File: entries/SWC-135.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Title
+Statements With No Effects
+
+## Relationships
+[CWE-561: Dead Code](https://cwe.mitre.org/data/definitions/561.html)
```
Comment (thec00n) [inline entries/SWC-135.md]: Is https://cwe.mitre.org/data/definitions/1164.html a more accurate CWE relationship?

File: entries/SWC-135.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Title
+Statements With No Effects
```
Comment (thec00n) [inline entries/SWC-135.md]: Why only statements? You could have expressions that have no effects as well would they fall under a different category? Also let's consider declarations.

File: entries/SWC-135.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+## Relationships
+[CWE-561: Dead Code](https://cwe.mitre.org/data/definitions/561.html)
+
+## Description
+In Solidity, it's possible to write statements that have no side effects. Currently, the solidity compiler will not return a warning for side-effect-free statements. This can lead to statements going unnoticed, causing a function to proceed without properly performing an intended action.
```
Comment (thec00n) [inline entries/SWC-135.md]: In which language is it not possible to do that, just curious.

File: entries/SWC-135.md (hunk 1 above)
Comment (KadenZipfel) [reply entries/SWC-135.md]: I think so.

File: entries/SWC-135.md (hunk 2 above)
Comment (KadenZipfel) [reply entries/SWC-135.md]: Good point. Perhaps just, "Code with no effects".

File: entries/SWC-135.md (hunk 3 above)
Comment (KadenZipfel) [reply entries/SWC-135.md]: I'm not sure of any in particular.

File: entries/SWC-134.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Title
+Code With No Effects
+
+## Relationships
+[CWE-1164: Irrelevant Code](https://cwe.mitre.org/data/definitions/1164.html)
+
+## Description
+In Solidity, it's possible to write code that has no side effects. Currently, the solidity compiler will not return a warning for side-effect-free code. This can lead to code going unnoticed, causing a function to proceed without properly performing an intended action.
```
Comment (b-mueller) [inline entries/SWC-134.md]: Should be just "effects" instead of "side effects". Not having side effects is usually a good thing?

Comment (b-mueller) [reply entries/SWC-134.md]: Or write,"it is possible to write code that does not produce the intended effect, for example by introducing subtle syntax errors. Currently, the Solidity compiler will usually not warn about such errors."

**Pull request closed without merging**